zallet/src/config.rs

```diff
@@ -233,3 +236,69 @@ impl RpcSection {
         Duration::from_secs(self.timeout.unwrap_or(30))
     }
 }
+
+impl ZalletConfig {
+    /// Generates an example config file, with all default values included as comments.
+    pub fn generate_example() -> String {
+        let mut config = r"# Default configuration for Zallet.
+#
+# This file is generated as an example using Zallet's current defaults. It can
+# be used as a skeleton for custom configs.
+#
+# Fields that are required to be set are uncommented, and set to an example
+# value. Every other field is commented out, and set to the current default
+# value that Zallet will use for it (or `UNSET` if the field has no default).
+#
+# Leaving a field commented out means that Zallet will always use the latest
+# default value, even if it changes in future. Uncommenting a field but keeping
+# it set to the current default value means that Zallet will treat it as a
+# user-configured value going forward.
+
+"
+        .to_owned();
+
+        fn write_section<T: Documented + Fields + DocumentedFields>(
+            config: &mut String,
+            section_name: &str,
+        ) {
+            writeln!(config).unwrap();
+            for line in T::DOCS.lines() {
+                writeln!(config, "# {}", line).unwrap();
+            }
+            writeln!(config, "[{}]", section_name).unwrap();
+            writeln!(config).unwrap();
+
+            for field_name in T::fields() {
+                write_field::<T>(config, field_name);
+            }
+        }
+
+        fn write_field<T: DocumentedFields>(config: &mut String, field_name: &str) {
+            let field_doc = T::get_field_docs(field_name).expect("present");
+            for line in field_doc.lines() {
+                writeln!(config, "# {}", line).unwrap();
+            }
+
+            // TODO: Get the default value for the field.
+            let field_default = toml::Value::try_from(vec![0u64]).unwrap();
```

Comment:
Currently we hard-code defaults inside getter methods, e.g. here:
https://github.com/zcash/wallet/blob/449eab1693b543bfcb9c1c4fac02eb866793cb38/zallet/src/config.rs#L130-L137

We instead need a way to map field names to default values, that we can share somehow between here and those getters.

Reply:
For now I've gone with a minimally-duplicating approach; we'll need to remember to update a list at the bottom of the file whenever we alter the structure of the config. But it's pretty short and should be manageable for now, and can be turned into a proc macro later if we want better UX.